### PR TITLE
[FrontEnd]Paddle_Op_Conversion_prior_box

### PR DESCRIPTION
**Pull request by liubo-intel**
### Details:
 - *PaddlePaddle op convert for "[prior_box"](https://www.paddlepaddle.org.cn/documentation/docs/en/1.7/api/layers/prior_box.html) , which is required by SSD-MobileNetV3*

### Tickets:
 - *63557*

### CODE REVIEW

**Comment by liubo-intel** [comment]
Hi, @ceciliapeng2011 @luo-cheng2021 @ilyachur @mvafin : 
could you please help review this PaddlePaddle Op conversion PR when you are free? tks~

File: ngraph/test/frontend/paddlepaddle/test_models/gen_scripts/generate_prior_box.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                                               max_sizes=attrs['max_sizes'],
+                                               aspect_ratios=attrs['aspect_ratios'],
+                                               variance=attrs['variance'],
+                                               flip=attrs['flip'],
+                                               clip=attrs['clip'],
+                                               steps=attrs['steps'],
+                                               offset=attrs['offset'],
+                                               name=None,
+                                               min_max_aspect_ratios_order=attrs['min_max_aspect_ratios_order'])
+
```
Comment (ceciliapeng2011) [inline ngraph/test/frontend/paddlepaddle/test_models/gen_scripts/generate_prior_box.py:31]: better fix the intent here

File: ngraph/frontend/paddlepaddle/src/op/prior_box.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/op/prior_box.hpp"
+
+#include <ngraph/opsets/opset8.hpp>
```
Comment (ceciliapeng2011) [inline ngraph/frontend/paddlepaddle/src/op/prior_box.cpp:7]: To align opset version, better follow this way for future op conversion -
https://github.com/openvinotoolkit/openvino/blob/b968c7b813a2e89ccd944f76d02b0ddd709fbc02/ngraph/frontend/paddlepaddle/src/op/gelu.cpp#L7

File: ngraph/frontend/paddlepaddle/src/op/prior_box.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+#include <node_context.hpp>
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+using namespace opset8;
```
Comment (ceciliapeng2011) [inline ngraph/frontend/paddlepaddle/src/op/prior_box.cpp:14]: the same

File: ngraph/frontend/paddlepaddle/src/op/prior_box.cpp (hunk 2 above)
Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/prior_box.cpp]: modified

File: ngraph/frontend/paddlepaddle/src/op/prior_box.cpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+using namespace default_opset;
+using namespace element;
+namespace detail {
+namespace {
+std::shared_ptr<StridedSlice> make_slice(std::shared_ptr<ngraph::Node> node, int64_t start, int64_t end) {
```
Comment (ilyachur) [inline ngraph/frontend/paddlepaddle/src/op/prior_box.cpp:19]:
```suggestion
std::shared_ptr<StridedSlice> make_slice(const std::shared_ptr<ngraph::Node>& node, int64_t start, int64_t end) {
```

File: ngraph/frontend/paddlepaddle/src/op/prior_box.cpp [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+    auto ov_prior_box_node = std::make_shared<PriorBox>(output_shape_slice, image_shape_slice, attrs);
+
+    auto split_axis_node = Constant::create(i64, ngraph::Shape{}, {0});
+    auto node_prior_box_split = std::make_shared<Split>(ov_prior_box_node, split_axis_node, 2);
+
+    auto node_boxes_origin = node_prior_box_split->output(0);
+    auto node_variances_origin = node_prior_box_split->output(1);
+
+    auto out_shape =
```
Comment (ilyachur) [inline ngraph/frontend/paddlepaddle/src/op/prior_box.cpp:57]: I think it can be `const auto`

**Review by mvafin — APPROVED** [review]

File: ngraph/frontend/paddlepaddle/src/op/prior_box.cpp (hunk 4 above)
Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/prior_box.cpp]: Done

File: ngraph/frontend/paddlepaddle/src/op/prior_box.cpp [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    bool min_max_aspect_ratios_order = node.get_attribute<bool>("min_max_aspect_ratios_order", false);
+
+    auto ov_prior_box_node = std::make_shared<PriorBox>(output_shape_slice, image_shape_slice, attrs);
+
+    auto split_axis_node = Constant::create(i64, ngraph::Shape{}, {0});
+    auto node_prior_box_split = std::make_shared<Split>(ov_prior_box_node, split_axis_node, 2);
+
+    auto node_boxes_origin = node_prior_box_split->output(0);
+    auto node_variances_origin = node_prior_box_split->output(1);
```
Comment (ilyachur) [inline ngraph/frontend/paddlepaddle/src/op/prior_box.cpp:55]: Majority of `auto` variables can be const

Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/prior_box.cpp]: sorry that misunderstood the previous review comments, have modified the left auto variables.

**Review by jane-intel — APPROVED** [review]
Miss comments in the code, but it looks good to me.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**